dotnet/src/Extensions/Experimental/Orchestration.Flow/FlowExecutor/FlowStatusProvider.cs

```diff
@@ -30,11 +31,13 @@ public FlowStatusProvider(IMemoryStore memoryStore, string? collectionName = nul
     {
         this._memoryStore = memoryStore;
         this._collectionName = collectionName ?? nameof(FlowStatusProvider);
-
-        if (!this._memoryStore.DoesCollectionExistAsync(this._collectionName).Result)
+        Task.Run(async () =>
         {
-            Task.Run(async () => await this._memoryStore.CreateCollectionAsync(this._collectionName).ConfigureAwait(false));
-        }
+            if (!await this._memoryStore.DoesCollectionExistAsync(this._collectionName).ConfigureAwait(false))
+            {
+                await this._memoryStore.CreateCollectionAsync(this._collectionName).ConfigureAwait(false);
+            }
+        }).GetAwaiter().GetResult();
```

Comment:
Consider extracting the code into a private method that will be called from all the methods that need the collection. An optimization (private field/flag) can be put in place to prevent unnecessary checks after the first one is done.

Reply:
Yeah I've moved this into an initialization method and made the constructor private and exposed a public `ConnectAsync` method for initialization. There are other ways this could be done, too.